infra/utils.sh

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function allow-sa-permissions() {
+    if [ $# -lt 1 ] && [ $# -gt 2 ] && [ -z "$1" ]; then
+        printf "allow-sa-permissions(project, account) requires 1 or 2 arguments" >&2
+        return 1
+    fi
+    local project="$1"
+    local account="$2"
+    local full_name_account=$(svc_acct_for "${project}" "${account}")
+    gcloud projects add-iam-policy-binding "${project}" \
+        --member "serviceAccount:${full_name_account}" \
+        --role roles/logging.logWriter
+    gcloud projects add-iam-policy-binding "${project}" \
+        --member "serviceAccount:${full_name_account}" \
+        --role roles/monitoring.viewer
+    gcloud projects add-iam-policy-binding "${project}" \
+        --member "serviceAccount:${full_name_account}" \
+        --role roles/monitoring.metricWriter
+}
+
+# Ensure the program is installed
+function check-install() {
+    if ! command -v "${1}" &>/dev/null; then
+        color 4 "You don't have ${1} installed. Please install ${1}."
+        exit 1
+    fi
+}
+
+# Create a BiqQuery Dataset for usage metering
+# $1 : The GCP project name
+# $2 : The BigQuery dataset name
+# $3 : The dataset region (optional)
+function create-bigquery-dataset() {
```

Comment:
Applies to all functions - please use shell_style_names.  Dashes in names confuse editors, and I can't fathom why it is allowed.

Reply:
This function should be more or less identical to the impl in `ensure-main-project.sh`, just wrapped into a function.  That impl includes setting permissions